src/datadog.ts

```diff
@@ -42,36 +42,55 @@ export function getClient(apiKey: string): httpm.HttpClient {
   })
 }
 
+async function postMetrics(
+  http: httpm.HttpClient,
+  apiURL: string,
+  metrics: {series: Array<Record<string, unknown>>},
+  endpoint: string
+): Promise<void> {
+  // POST data
+  if (metrics.series.length) {
+    core.debug(`About to send ${metrics.series.length} metrics`)
+    const res: httpm.HttpClientResponse = await http.post(
+      `${apiURL}/api/${endpoint}`,
+      JSON.stringify(metrics)
+    )
+
+    if (res.message.statusCode === undefined || res.message.statusCode >= 400) {
+      throw new Error(
+        `HTTP request failed: ${res.message.statusMessage} ${res.message.statusCode}`
+      )
+    }
+  }
+}
+
 export async function sendMetrics(
   apiURL: string,
   apiKey: string,
   metrics: Metric[]
 ): Promise<void> {
   const http: httpm.HttpClient = getClient(apiKey)
-  const s = {series: Array()}
+  // distributions use a different procotol.
+  const distributions = {series: Array()}
+  const otherMetrics = {series: Array()}
   const now = Date.now() / 1000 // timestamp must be in seconds
 
   // build series payload containing our metrics
   for (const m of metrics) {
-    s.series.push({
+    const isDistribution = m.type === 'distribution'
+    const value = isDistribution ? [m.value] : m.value
+    const collector = isDistribution ? distributions : otherMetrics
+    collector.series.push({
       metric: m.name,
-      points: [[now, m.value]],
+      points: [[now, value]],
       type: m.type,
       host: m.host,
       tags: m.tags
     })
   }
 
-  // POST data
-  core.debug(`About to send ${metrics.length} metrics`)
-  const res: httpm.HttpClientResponse = await http.post(
-    `${apiURL}/api/v1/series`,
-    JSON.stringify(s)
-  )
-
-  if (res.message.statusCode === undefined || res.message.statusCode >= 400) {
-    throw new Error(`HTTP request failed: ${res.message.statusMessage}`)
-  }
+  await postMetrics(http, apiURL, otherMetrics, 'v1/series')
+  await postMetrics(http, apiURL, distributions, 'v1/distribution_points')
```

Comment:
Do you think it makes sense to skip the call in case there are not metrics of a certain type? Say something like
```ts
if (otherMetrics.length > 0)
  await postMetrics(http, apiURL, otherMetrics, 'v1/series')

if (distributions.length > 0) 
  await postMetrics(http, apiURL, distributions, 'v1/distribution_points')
```

Reply:
yes I do - that [is coded for](https://github.com/masci/datadog/pull/25/files#diff-75a093847e23e633ee899d98d0b61b2a0ed87d61004a1ae60ac3bfc083eefa3fR52). Perhaps I should change the name of the supporting method to make it clearer from `postMetrics` to `postMetricsIfAny`

Reply:
We are currently running this version with only one distribution metric, and I believe it is skipping the other metrics.

Reply:
Ah great, I totally missed that, thanks for renaming the method!